pkg/kinds/convert_test.go

```diff
@@ -0,0 +1,540 @@
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolTCP,
+							Port:       int32(80),
+							TargetPort: intstr.FromInt(9376),
+						},
+					},
+				},
+			},
+			scheme: coreScheme,
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": Service().GroupVersion().String(),
+					"kind":       Service().Kind,
+					"metadata": map[string]interface{}{
+						"name":              "test-name",
+						"creationTimestamp": nil, // Added field
+					},
+					"spec": map[string]interface{}{
+						"selector": map[string]interface{}{
+							"app.kubernetes.io/name": "MyApp",
+						},
+						"ports": []interface{}{
+							map[string]interface{}{
+								"protocol":   "TCP",
+								"port":       int64(80),   // Type change
+								"targetPort": int64(9376), // Type change
+							},
+						},
+					},
+					"status": map[string]interface{}{ // Added field
+						"loadBalancer": map[string]interface{}{}, // Added field
+					},
+				},
+			},
+		},
+		{
+			name: "typed unpopulated GVK not in scheme",
+			object: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-name",
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"app.kubernetes.io/name": "MyApp",
+					},
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolTCP,
+							Port:       int32(80),
+							TargetPort: intstr.FromInt(9376),
+						},
+					},
+				},
+			},
+			scheme: emptyScheme,
+			expectedError: testutil.EqualError(
+				errors.Wrap(
+					runtime.NewNotRegisteredErrForType(emptyScheme.Name(),
+						reflect.TypeOf(corev1.Service{})),
+					"failed to lookup object type")),
+		},
+		{
+			name: "typed unpopulated GVK in scheme",
+			object: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-name",
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"app.kubernetes.io/name": "MyApp",
+					},
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolTCP,
+							Port:       int32(80),
+							TargetPort: intstr.FromInt(9376),
+						},
+					},
+				},
+			},
+			scheme: coreScheme,
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": Service().GroupVersion().String(),
+					"kind":       Service().Kind,
+					"metadata": map[string]interface{}{
+						"name":              "test-name",
+						"creationTimestamp": nil, // Added field
+					},
+					"spec": map[string]interface{}{
+						"selector": map[string]interface{}{
+							"app.kubernetes.io/name": "MyApp",
+						},
+						"ports": []interface{}{
+							map[string]interface{}{
+								"protocol":   "TCP",
+								"port":       int64(80),   // Type change
+								"targetPort": int64(9376), // Type change
+							},
+						},
+					},
+					"status": map[string]interface{}{ // Added field
+						"loadBalancer": map[string]interface{}{}, // Added field
```

Comment:
Let us add the discussion in https://github.com/GoogleContainerTools/kpt-config-sync/pull/77#discussion_r961950738 to here as a comment to help others understand this.

Reply:
done